adapters/exco/exco.go

```diff
@@ -0,0 +1,185 @@
+
+	reqData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqjsonutil,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{reqData}, errs
+}
+
+func (a *adapter) MakeBids(
+	internalRequest *openrtb2.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	var errs []error
+	bidderResponse := adapters.NewBidderResponse()
+	for _, seatBid := range bidResponse.SeatBid {
+		for i := range seatBid.Bid {
+			bid := &seatBid.Bid[i]
+
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	return bidderResponse, errs
+}
+
+// getMediaTypeForBid determines which type of bid.
+func getMediaTypeForBid(bid *openrtb2.Bid) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case 1:
+		return openrtb_ext.BidTypeBanner, nil
+	case 2:
+		return openrtb_ext.BidTypeVideo, nil
+	default:
+		return "", fmt.Errorf("unrecognized bid_ad_type in response from exco: %d", bid.MType)
+	}
+}
+
+func adjustRequest(
+	request *openrtb2.BidRequest,
+) (*openrtb2.BidRequest, error) {
+	var publisherId string
+
+	for i := 0; i < len(request.Imp); i++ {
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(request.Imp[i].Ext, &bidderExt); err != nil {
+			return nil, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext for impression index %d. Error Information: %s", i, err.Error()),
+			}
+		}
+
+		var impExt openrtb_ext.ImpExtExco
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+			return nil, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext.bidder for impression index %d. Error Information: %s", i, err.Error()),
+			}
+		}
+
+		if impExt.PublisherId == "" {
+			return nil, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext.bidder for impression index %d. Error Information: %s", i, "Missing publisherId"),
+			}
+		}
+
+		if impExt.TagId == "" {
+			return nil, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext.bidder for impression index %d. Error Information: %s", i, "Missing tagId"),
+			}
+		}
+
+		if impExt.AccountId == "" {
+			return nil, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext.bidder for impression index %d. Error Information: %s", i, "Missing accountId"),
+			}
+		}
+
+		publisherId = impExt.PublisherId
```

Comment:
`publisherId` variable will have a value from the last process impression. Is this expected?